### PR TITLE
chore(Bonus Pagamenti Digitali): [IAC-129] Implemented payment methods selection screen

### PR DESCRIPTION
**Pull request by dgopsq**
## Short description
This PR is going to add the payment methods selection screen for the opt-in payment methods flow.

https://user-images.githubusercontent.com/5963683/154653802-f7ac046f-8800-495c-a3a3-d039b8f82238.mov

### N.B.
- The `Consigliato` badge was not translated in the document, I added a `Suggested` translation but it should be reviewed.

## List of changes proposed in this pull request
- Implemented the `OptInPaymentMethodsChoiceScreen`.
- Created a `IOBadge` centralized component.
- Added a prop `rightSideSelection` to the `RadioButtonList` component. This is used to align to the right the radio buttons.

## How to test
Enable `bpdOptInPaymentMethodsEnabled` and dispatch `navigation.dispatch(navigateToOptInPaymentMethodsChoiceScreen())` to reach the screen.

### CODE REVIEW

File: locales/en/index.yml [hunk 1]
```diff
@@ -2533,8 +2533,18 @@ bonus:
         subtitle: "Ministry of Economy and Finance"
         body: !include bonus/bpd/optInPaymentMethods.md
       choice:
-        header: "Configura IO"
-        title: "Vuoi mantenere i tuoi metodi di pagamento?"
+        header: Configure IO
+        title: Do you want to keep your payment methods?
+        subtitle: If you keep them, you’ll be able to use them to take part in discounts, bonuses and future initiatives or for other features, such as in-app payments.
+        suggestedOption: Suggested
```
Comment (thisisjp) [inline locales/en/index.yml:2539]:
```suggestion
        suggestedOption: Recommended
```

File: ts/features/bonus/bpd/screens/optInPaymentMethods/OptInPaymentMethodsChoiceScreen.tsx [hunk 2]
```diff
@@ -1,36 +1,160 @@
...
+) => (
+  <>
+    {showBadge ? (
+      <>
+        <IOPrimaryBadge>
+          {I18n.t("bonus.bpd.optInPaymentMethods.choice.suggestedOption")}
+        </IOPrimaryBadge>
+        <View spacer xsmall />
+      </>
+    ) : undefined}
```
Comment (Undermaken) [inline ts/features/bonus/bpd/screens/optInPaymentMethods/OptInPaymentMethodsChoiceScreen.tsx:48]:
```suggestion
    {showBadge && (
      <>
        <IOPrimaryBadge
          text={I18n.t("bonus.bpd.optInPaymentMethods.choice.suggestedOption")}
        />
        <View spacer xsmall />
      </>
    )}
```
I guess conditional render with `&&` is more readable when you have not an `else` case

File: ts/components/core/selection/RadioButtonList.tsx [hunk 3]
```diff
@@ -17,6 +17,12 @@ type Props<T> = {
   head?: string;
   selectedItem?: T;
   onPress: (selected: T) => void;
+
+  /**
+   * This prop will align the radio buttons
+   * to the right and the label to the left.
+   */
+  rightSideSelection?: boolean;
```
[the remark this answers is not in the document]
Comment (dgopsq) [reply ts/components/core/selection/RadioButtonList.tsx:25]: It's really rare, given the component type, but I can image a situation in which you want that prop to be dynamically assigned via a boolean variable, in that case having that prop typed as `true | undefined` could be a discomfort, what do you think? Or maybe I'm missing something?

File: ts/features/bonus/bonusVacanze/components/buttons/ButtonConfigurations.ts [hunk 4]
```diff
@@ -56,17 +57,41 @@ export const errorButtonProps = (
   onPress
 });
 
+/**
+ * Style for error props
+ * @param onPress
+ * @param title
+ */
+export const errorBorderedButtonProps = (
```
[the remark this answers is not in the document]
Comment (dgopsq) [reply ts/features/bonus/bonusVacanze/components/buttons/ButtonConfigurations.ts:65]: Yeah, I was thinking the same while adding this function 😂 Maybe we should open an infrastructure ticket aiming to move this component?

File: ts/components/core/selection/RadioButtonList.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Undermaken) [reply ts/components/core/selection/RadioButtonList.tsx:25]: You are right!
I didn't spot this side!
Let's keep as it is

File: ts/features/bonus/bonusVacanze/components/buttons/ButtonConfigurations.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (Undermaken) [reply ts/features/bonus/bonusVacanze/components/buttons/ButtonConfigurations.ts:65]: Yes, good points, since it could impact few files

**Review by Undermaken — APPROVED** [review]

**Pull request merged**